client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts

```diff
@@ -261,12 +268,19 @@ function getFallbackDestination( {
 		if ( isJetpackNotAtomic && purchasedProduct ) {
 			debug( 'the site is jetpack and bought a jetpack product', siteSlug, purchasedProduct );
 
+			// Jetpack Cloud will either redirect to wp-admin (if JETPACK_CLOUD_REDIRECT_CHECKOUT_TO_WPADMIN
+			// flag is set), or otherwise will redirect to a Jetpack Redirect API url (source=jetpack-checkout-thankyou)
 			if ( isJetpackCloud() ) {
+				if ( redirectCloudCheckoutToWpAdmin() && adminUrl ) {
+					debug( 'checkout is Jetpack Cloud, returning wp-admin url' );
+					return `${ adminUrl }admin.php?page=jetpack#/my-plan`;
+				}
 				debug( 'checkout is Jetpack Cloud, returning Jetpack Redirect API url' );
-				return `${ JETPACK_REDIRECT_URL }?source=jetpack-checkout-thankyou&site=${ siteSlug }&query=${ encodeURIComponent(
+				return `${ JETPACK_REDIRECT_URL }&site=${ siteSlug }&query=${ encodeURIComponent(
```

Comment:
[Non-blocking] We'll need to be careful of this ampersand in the future if `JETPACK_REDIRECT_URL` ever loses its initial query parameter.